compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -62,14 +62,15 @@ pub fn ensure_removed(diag_handler: &Handler, path: &Path) {
 /// of the requested outputs for this compilation session.
 pub fn link_binary<'a>(
     sess: &'a Session,
+    crate_types: &[CrateType],
```

Comment:
Maybe move this into CodegenResults? That will make `-Zlink-only` correctly handle `#![crate_type]` by serializing the crate type in the `.rlink` file.

Reply:
This looks like the right thing to do, implemented it.
The previous version was pretty mechanical.